### DISCUSSION TITLE
how to open file in new tab but not navigate to it

### DISCUSSION THREAD

**Discussion by holynewbie** [microsoft/vscode-discussions#584] [Extension Development QnA] [answered]
My extension wants to open a new file in new tab, but stay focus on current tab. I used code below but not work.

```ts
const uri = vscode.Uri.file('some path');
const doc = await workspace.openTextDocument(uri)
await window.showTextDocument(doc, {
        preview: true,
        preserveFocus: true
});
```

it opens the file and navigate to it.

what i want is just like open link in the browser

https://user-images.githubusercontent.com/8347286/234550446-5ffd52ef-c19d-4833-ae6f-5deac1625b93.mp4

**Comment by alefragnani** [comment, marked as answer]
Hi @holynewbie ,

The current API does not support this. It will always open and show the document to the user.

After a quick look at VS Code repo, I have found [this internal request](https://github.com/microsoft/vscode/issues/96964), which has made a similar request. But after a few discussions in the PR (https://github.com/microsoft/vscode/pull/96966#issuecomment-642285859), ended up reverting the API change and instead, updating the commands which needed that behavior.

The same PR says the `vscode.open` command has been updated to support a `background` option. So, you could alternatively do this:

```ts
    vscode.commands.executeCommand("vscode.open", fileUri, { background: true });
```


Hope this helps

**Reply by holynewbie** [reply]
tks a lot, it does help. 😊
